### ISSUE TITLE
InlineData should use InvariantCulture to convert arguments

### ISSUE THREAD

**Issue by CornedBee** [xunit/xunit#1337]
TL;DR: `Xunit.Sdk.Reflector.ConvertArgument` should pass `CultureInfo.InvariantCulture` to `Convert.ChangeType`.

A simple test case that looks like this:

```
[Theory]
[InlineData(false, "1.0")]
void DecimalsGreaterFive(bool expected, decimal value)
{
  Assert.Equal(expected, value > 5);
}
```

fails or misbehaves depending on the culture settings of the computer on which the test case runs. If the culture uses '.' as a decimal separator (e.g. 'en-US'), it succeeds. If it uses '.' as a thousands separator (e.g. 'de-DE'), it fails because it parses the number as 10. If it doesn't allow '.' at all (e.g. 'de-AT', where the comma is the decimal separator and the space is the thousands separator), it throws an exception in the test runner infrastructure.

I believe test cases should not depend on the settings of the computer that is executing them. Working around this problem is tricky - for example, setting the current thread culture in the constructor doesn't work, because the method argument conversion happens before the test object is instantiated. (I think, but I am not sure, that it also happens before a BeforeAfterMethodAttribute is executed, so the UseCultureAttribute from the samples wouldn't work either.)

The working workaround (on .Net Core at least) is to create a class fixture that changes `CultureInfo.DefaultThreadCurrentCulture`. Setting `CultureInfo.CurrentCulture` doesn't work; I suspect that the culture of some other thread flows along the async test executor into the current culture of the test case after it was changed in the fixture constructor. (This may mean that the behavior is different in .Net < 4.6 where the culture doesn't flow.)

This could be easily avoided if the test runner did its conversions with a fixed culture. There is no reason to use the current culture here, and no reason why anyone would want the InlineData values to be localized. So the solution would be to just pass `CultureInfo.InvariantCulture` to `Convert.ChangeType` in `Xunit.Sdk.Reflector.ConvertArgument` (src/xunit.execution/Sdk/Reflection/Reflector.cs). This converts the arguments ahead of time to the correct types with a fixed culture.

**Comment by nohwnd** [comment]
We just hit this issue, and I fully support using `InvariantCulture` for the conversion.

**Comment by bradwilson** [comment]
This has been fixed.

**Comment by nohwnd** [comment]
@bradwilson Thank you.
